handle_pay_keysend_request.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/nbd-wtf/go-nostr"
+	"github.com/sirupsen/logrus"
+)
+
+func (svc *Service) HandlePayKeysendEvent(ctx context.Context, request *Nip47Request, event *nostr.Event, app App, ss []byte) (result *nostr.Event, err error) {
+
+	nostrEvent := NostrEvent{App: app, NostrId: event.ID, Content: event.Content, State: "received"}
+	err = svc.db.Create(&nostrEvent).Error
+	if err != nil {
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+			"appId":     app.ID,
+		}).Errorf("Failed to save nostr event: %v", err)
+		return nil, err
+	}
+
+	payParams := &Nip47KeysendParams{}
+	err = json.Unmarshal(request.Params, payParams)
+	if err != nil {
+		svc.Logger.WithFields(logrus.Fields{
+			"eventId":   event.ID,
+			"eventKind": event.Kind,
+			"appId":     app.ID,
+		}).Errorf("Failed to decode nostr event: %v", err)
+		return nil, err
+	}
+
+	hasPermission, code, message := svc.hasPermission(&app, event, request.Method, 0)
+	if hasPermission {
+		// We use pay_invoice permissions for budget and max amount
```

Comment:
Why not use the keysend permission?

Reply:
https://github.com/getAlby/nostr-wallet-connect/pull/178#discussion_r1425060561